src/luarocks/cmd.lua

```diff
@@ -306,6 +304,128 @@ do
+   for _, tree in ipairs(cfg.rocks_trees) do
+      if type(tree) == "string" then
+         buf = buf.."      "..fs.absolute_name(tree)
+      else
+         local name = tree.name and " (\""..tree.name.."\")" or ""
+         buf = buf.."      "..fs.absolute_name(tree.root)..name
+      end
+   end
+
+   return buf.."\n"
+end
+
+local function get_parser(description, cmd_modules)
+   local basename = dir.base_name(program)
+   local parser = argparse(
+      basename, "LuaRocks "..cfg.program_version..", the Lua package manager\n\n"..
+      program.." - "..description)
+      :help_max_width(80)
+      :add_help("--help")
+      :add_help_command()
+      :add_complete_command({
+         help_max_width = 120,
+         summary = "Output a shell completion script.",
+         description = [[
+Output a shell completion script.
+
+Enabling completions for Bash:
+
+   Add the following line to your ~/.bashrc:
+      source <(]]..basename..[[ completion bash)
+   or save the completion script to the local completion directory:
+      ]]..basename..[[ completion bash > ~/.local/share/bash-completion/completions/]]..basename..[[
+
+
+Enabling completions for Zsh:
+
+   Save the completion script to a file in your $fpath.
+   You can add a new directory to your $fpath by adding e.g.
+      fpath=(~/.zfunc $fpath)
+   to your ~/.zshrc.
+   Then run:
+      ]]..basename..[[ completion zsh > ~/.zfunc/_]]..basename..[[
+
+
+Enabling completion for Fish:
+
+   Add the following line to your ~/.config/fish/config.fish:
+      ]]..basename..[[ completion fish | source
+   or save the completion script to the local completion directory:
+      ]]..basename..[[ completion fish > ~/.config/fish/completions/]]..basename..[[.fish
+]]})
+      :command_target("command")
+      :require_command(false)
+
+   parser:flag("--version", "Show version info and exit.")
+      :action(function()
+         util.printout(program.." "..cfg.program_version)
+         util.printout(description)
+         util.printout()
+         os.exit(cmd.errorcodes.OK)
+      end)
+   parser:flag("--dev", "Enable the sub-repositories in rocks servers for "..
+      "rockspecs of in-development versions.")
+   parser:option("--server", "Fetch rocks/rockspecs from this server "..
+      "(takes priority over config file).")
+   parser:option("--only-server", "Fetch rocks/rockspecs from this server only "..
+      "(overrides any entries in the config file).")
+      :argname("<server>")
+   parser:option("--only-sources", "Restrict downloads to paths matching the given URL.")
+      :argname("<url>")
+   parser:option("--namespace"):description("Specify the rocks server namespace to use.")
+   parser:option("--lua-dir", "Which Lua installation to use.")
+      :argname("<prefix>")
+   parser:option("--lua-version", "Which Lua version to use.")
+      :argname("<ver>")
+   parser:option("--tree", "Which tree to operate on.")
+   parser:option("--project-tree"):description("") -- TODO
```

Comment:
Not sure what the description should be for this option.

Reply:
(I haven't looked properly at this branch yet, but I noticed this comment in a quick fly-by)

`--project-tree` is a slightly "internal" option, as it is used by the `./luarocks` wrapper created by `luarocks init` to basically force the use of the particular project tree, regardless of what the current directory is. Rationale:

* plain `luarocks` works like `git`: depending on your working directory, it works on a given project
* but if I have `~/project_a` and `~/project_b`, both initalized with `luarocks init`, and I do `cd ~/project_b; ../project_a/luarocks`, I would expect that wrapper to still refer to `project_a`.

(Having said that, as we are working out the kinks of running `luarocks` reliably in a `git`-style fashion, I'm getting less confident about the necessity of the `./luarocks` wrapper — I still think the `./lua` wrapper is more useful, if only to avoid having to deal with package paths (Lua package path errors are the kind of thing that are easy once you know them, but are a total mistery for newcomers))